things/database.py

```diff
@@ -685,28 +685,38 @@ def make_filter(column, value):
     }.get(value, default)
 
 
-def make_date_filter(create_var_name, last):
-    """Returns SQL filter to limit date range of the SQL query.
+def make_date_filter(created_column, offset):
+    """
+    Returns SQL filter to limit date range of the SQL query.
+    """
+
+    # Offset not specified
+    if offset is None:
+        return ""
 
-    Args:
-        create_var_name (string): Name of the column that has information on when each task was created.
-        last (string): A string comprised of an integer and a single character that can be 'd' or 'w' that determines whether to return all tasks for the past X days or weeks.
+    # Validation
 
-    Returns:
-        string: A string to add into the filter for the SQL query.
-    """
-    if (type(last) is str):
-        if last[-1] == "d":
-            return f'AND datetime(TASK.{create_var_name}, "unixepoch", "localtime") BETWEEN datetime("now", "-{int(last[:-1])} days") AND datetime("now", "localtime")'
-        elif last[-1] == "w":
-            return f'AND datetime(TASK.{create_var_name}, "unixepoch", "localtime") BETWEEN datetime("now", "-{int(last[:-1])*7} days") AND datetime("now", "localtime")'
-        else:
-            print("Please specify last as a string of the format 'Xd' where X is a non-negative integer and 'd' states that it is measured in days")    
-    else:
-        if last:
-            print("Please specify last as a string of the format 'Xd' where X is a non-negative integer and 'd' states that it is measured in days")    
-        else:
-            return ""
+    if not isinstance(offset, str):
+        raise ValueError(
+            "Please specify last as a string of the format 'X[d/w]' where X is a non-negative integer followed by 'd' or 'w' that indicates days or weeks."
+        )
+
+    suffix = offset[-1]
+    if suffix not in ("d", "w"):
+        raise ValueError(
+            "Please specify last as a string of the format 'X[d/w]' where X is a non-negative integer followed by 'd' or 'w' that indicates days or weeks."
```

Comment:
If you want to output the same error message, I'd wrap the two checks into one like so:

```python
if not isinstance(offset, str) or offset[-1] not in ("d", "w"):
    ValueError(...)
```

Reply:
The philosophy I use here is "don't repeat yourself" or "don't copy paste" :) — It makes maintenance easier down the road when things change.

Reply:
Also, maybe the error message could refer to the parameter as `offset`?

For example:

```python
ValueError('Invalid offset: {offset!r}.\nA valid offset is a string of …')
```